### ISSUE TITLE
FV header doesn't validate the checksum

### ISSUE THREAD

**Issue by gibarrasch** [chipsec/chipsec#480]
Hi Guys, I noticed that the header validation of the FV is disabled. is there any reason?

https://github.com/chipsec/chipsec/blob/master/chipsec/hal/uefi_common.py#L579

**Comment by ErikBjorge** [comment]
I agree that we should validate the checksum and produce a warning if it is bad.

**Comment by 0xfede7c8** [comment]
I have another question about this. Most of the times the check is disabled by the header Attributes.
Only if the attributes have the FFS_ATTRIB_CHECKSUM flag, the checksum is calculated and compared. If not, it is hardcoded to FFS_FIXED_CHECKSUM.

Why does this happens?

**Comment by skochinsky** [comment]
This is a question to UEFI spec writers. At a guess, checksum calculation over the whole file can be  expensive so it was made optional and is used sparingly. Besides, the 8-bit sum of bytes is not very robust and may fail in very simple  corruption cases. Some vendors use a CRC32 wrapper section when they need a reliable integrity check of the file's data.

**Comment by 0xfede7c8** [comment]
Great thanks!
